core/object/object.cpp

```diff
@@ -1033,14 +1034,33 @@ Error Object::emit_signalp(const StringName &p_name, const Variant **p_args, int
 
 		const Variant **args = p_args;
 		int argc = p_argcount;
+		Callable ca = c.callable;
+
+		if (c.flags & CONNECT_APPEND_SOURCE_OBJECT) {
+			if (unlikely(appended_self.get_type() == Variant::NIL)) {
+				appended_self = this;
+			}
+
+			if (ca.is_custom() && dynamic_cast<CallableCustomUnbind *>(ca.get_custom())) {
```

Comment:
I think handling unbind should not be done, this is up to the user to deal with.